### PR TITLE
Make loopback2 use tapDevice from l2.

### PR DESCRIPTION
**Pull request by djpetti**
In theory, this shouldn't really change anything...
However, I'm sort of working in the dark here, so I wanted to make sure that I didn't do anything stupid before I go and add latency.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -6,32 +6,72 @@ package main
...
+// Tap devices need to be cleaned up when we exit. This function gets run in a
+// goroutine and handle it.
+// Args:
+//	signalChannel: Channel we're receiving the signal on.
+//	cleanupDone: Channel we use to notify the main thread that we're done
+//	cleaning up.
+//	devices: Slice of devices to clean up.
+func handleCleanup(signalChannel <-chan os.Signal, cleanupDone chan<- bool,
+	devices *[]l2.FrameReadWriteCloser) {
+	for _ = range signalChannel {
```
Comment (c00w) [inline main.go]: Just do <- signalChannel

File: main.go [hunk 2]
```diff
@@ -6,32 +6,72 @@ package main
...
+}
+
+// Tap devices need to be cleaned up when we exit. This function gets run in a
+// goroutine and handle it.
+// Args:
+//	signalChannel: Channel we're receiving the signal on.
+//	cleanupDone: Channel we use to notify the main thread that we're done
+//	cleaning up.
+//	devices: Slice of devices to clean up.
+func handleCleanup(signalChannel <-chan os.Signal, cleanupDone chan<- bool,
```
Comment (c00w) [inline main.go]: Don't have a cleanup channel, just block the main goroutine.

File: main.go [hunk 3]
```diff
@@ -6,32 +6,72 @@ package main
...
 			log.Fatal("Error writing packet:", err)
 		}
 	}
 }
 
+// Generates MAC addresses for each device.
+// Args:
+//	device_number: Which device we're currently working with. This is just here
+//	to make sure that all our MAC addresses are unique.
+func generateMac(device_number int64) string {
```
Comment (c00w) [inline main.go]: Might be easier to just call random but *shrug*

File: main.go [hunk 4]
```diff
@@ -70,43 +114,66 @@ func main() {
...
+
 				go dump(lo0, lo1)
 				go dump(lo1, lo0)
 			}
 		}
-		<-quitChannel
 
 	} else { // set up the testing as normal
-		lo0, err := tuntap.Open("looptap%d-0", tuntap.DevTap)
+		var device_number int64 = 0
```
Comment (c00w) [inline main.go]: device_number := int64(0)?

File: main.go [hunk 5]
```diff
@@ -6,32 +6,72 @@ package main
...
 
 	"os"
+	"os/signal"
 )
 
 var config = flag.String("config", "",
 	"A configuration file for the network")
 
-func dump(src, dest *tuntap.Interface) {
+func dump(src, dest l2.FrameReadWriteCloser) {
```
Comment (c00w) [inline main.go]: This whole function can just be replaced with l2.SendFrames

**Comment by c00w** [comment]
Looks mostly good. Fix the minor cleanups + make sure that it still works then merge it?

File: main.go (hunk 1 above)
Comment (djpetti) [inline main.go]: Done.

File: main.go (hunk 2 above)
Comment (djpetti) [inline main.go]: Done.

File: main.go (hunk 3 above)
Comment (djpetti) [inline main.go]: Well, yes, but I'm too OCD to leave the possibility of getting duplicates open...

File: main.go (hunk 4 above)
Comment (djpetti) [inline main.go]: Done.

File: main.go (hunk 5 above)
Comment (djpetti) [inline main.go]: Ooh, good point. I missed that.

**Pull request merged**